lib/api/controllers/remoteActions/swagger.js

```diff
@@ -0,0 +1,158 @@
+var
+  fs = require('fs'),
+  _ = require('lodash'),
+  path = require('path'),
+  InternalError = require('kuzzle-common-objects').Errors.internalError,
+  q = require('q'),
+  jsonToYaml = require('json2yaml'),  
+  generateSwagger,
+  writeFiles;
+
+generateSwagger = (routes, apiVersion) => {
+  var swagger;
+
+  swagger = {
+    basePath: '/api',
+    consumes: [
+      'application/json'
+    ],
+    host: 'sandbox.kuzzle.io:7511',
+    info: {
+      contact: {
+        email: 'hello@kuzzle.io',
+        name: 'Kuzzle team',
+        url: 'http://kuzzle.io'
+      },
+      description: 'The Kuzzle REST API',
+      license: {
+        name: 'Apache 2',
+        url: 'http://opensource.org/licenses/apache2.0'
+      },
+      title: 'Kuzzle API',
+      version: apiVersion
+    },
+    produces: [
+      'application/json'
+    ],
+    schemes: [
+      'http'
+    ],
+    swagger: '2.0',    
+    paths: {}
+  };
+
+  routes.forEach(route => {
+    var 
+      reg = /:([^\/]*)/g, // since url parameters got the following form "/:parameterName/:otherParameter" this reg captures, all chars following : which are not a / char
+                          // the first and las / are reg delimiters
```

Comment:
nitpicking: well... the first comment line is **very** helpful, but the other ones are only explaining regular expressions, and can/should be removed